src/qibolab/backends.py

```diff
@@ -50,6 +52,32 @@ def __init__(self, platform):
         }
         self.compiler = Compiler.default()
 
+    @property
+    def qubits(self) -> list[Union[str, int]]:
+        return list(self.platform.qubits)
+
+    @property
+    def connectivity(self) -> list[tuple[Union[str, int], Union[str, int]]]:
+        return list(self.platform.pairs)
+
+    @property
+    def natives(self) -> list[str]:
+        native_gates = set()
+        for _, q in self.platform.qubits.items():
+            native_gates |= {
+                fld.name
```

Comment:
Here `fld.name` for single qubit gates will be `RX` and `MZ`. Are you sure that these are the names that must be communicated to qibo? 

Because `MZ` doesn't exist in qibo (we call it `M` there) and `RX` could be considered a native gate, but because it is not universal on its own (it also needs a phase), we are treating `GPI2` as the universal one which we compile to `RX` pulses (basically `RX90`) in the qibolab compiler. Therefore, as far as the qibo transpiler is concerned, I would expect that `GPI2` should be communicated as the native single-qubit gate (unless we want to compile some other gate in a particular application).

Reply:
Following this, I suspect that `QibolabBackend.natives` may also need to check what compiler rules are available to find the "native gates" (at least the single-qubit ones), instead of the platform itself. Note that this backend already has a `self.compiler` attribute, so you should be able to get the gates from `self.compiler.rules`. I would also check what others (@alecandido) think though.

Reply:
Nice catch @stavros11!

Yes, I definitely overlooked that, and you're right that even the compiler plays a role.

So, for the time being, in principle you could even avoid checking the platform for the natives, since we support a single compiler for all platforms. And the compiler is truly the relevant object you care about.

Thus, for the short term, you could return a fixed list containing something like `[GPI2, RZ, M, ...]` (inspecting the compiler), and possibly just check the platform to inspect which two-qubit gates are available (though it will be `CZ` most of the time).

Later on, we should work more on the compiler interface, and possibly associate specific compilers to the platforms. But this is not for the short term.

Reply:
Regarding the naming of gates, which is the better approach?

- restrict the use of native gate names on hardware backends ex) using `CX` instead of `CNOT`
- add parsing logic to qibo to standardize gate names ex) `CX`, `CNOT`, `cx`, `cnot` -> `CX`

Reply:
`qibo` is a dependency for `qibolab`, so that's the upstream, and it has to be Qibolab to follow Qibo's conventions.

Other than that, following what's written above, I'd suggest you to just convert the names used internally in Qibolab for two-qubit gates within this same method.